explorer/data/prelude.carbon

```diff
@@ -416,6 +416,56 @@ external impl i32 as RightShiftWith(i32) where .Result == i32 {
   }
 }
 
+//-------------------------
+// Optional
+//-------------------------
+choice OptionalElement(T:! Type) {
+  None(),
+  Element(T)
+}
+
+class Optional(T:! Type){
+  fn CreateEmpty() -> Optional(T){
+    return { .element = OptionalElement(T).None() };
+  }
+  fn Create ( value: T ) -> Optional(T){
+    return { .element = OptionalElement(T).Element(value) };
+  }
+
+  fn has_value[me: Self] () -> bool{
+    var x: OptionalElement(T) = me.element;
+    match(x){
```

Comment:
Can this be rewritten as:
```suggestion
    match (me.element){
```
... or does explorer reject that for some reason?

Reply:
Rewritten!